### PR TITLE
Feature/Bugfix: Simplified adders and policy state fix

### PR DESCRIPTION
**Pull request by DriesSmit**
## What?
Version 2 of the policy state mismatch PR.

- Fix the policy state mismatch bug.
- Update the adders to be closer to Acme's implementation by removing the `next_extras` variable.
- Remove support for multiple trainers and agent batching. This drastically simplifies the base adder code.

TODO:

- Fix the tests (with the help of @OmaymaMahjoub :slightly_smiling_face:)

## Why?
Fixes a bug in the recurrent policy implementation and simplifies the Mava codebase.
## How?
Updated the adder code and fixed a bug in the executor's observing component. The policy_state is now passed through the extras variable by delaying writing it to the adder by one step.
## Extra
Close #850

### CODE REVIEW

File: mava/adders/base.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ class ParallelAdder(adders.Adder):
     ```python
     # Reset the environment and add the first observation.
     timestep = env.reset()
-    adder.add_first(timestep.observation)
```
Comment (OmaymaMahjoub) [inline mava/adders/base.py:39]: Why is this edited?

Comment (DriesSmit) [reply mava/adders/base.py:39]: No idea :raised_eyebrow: Please change it back if you get the chance.

**Review by KaleabTessera — APPROVED** [review]
Thanks @DriesSmit ! This is epic! Much easier to understand now!! :rocket: :fire:

**Review by sash-a — APPROVED** [review]
LGTM :fire:. Tested on DQN branch and all seems in order.

**Pull request merged**